src/mono/sample/iOS-NativeAOT/Program.csproj

```diff
@@ -8,14 +8,19 @@
     <DeployAndRun Condition="'$(DeployAndRun)' == ''">true</DeployAndRun>
     <RuntimeIdentifier>$(TargetOS)-$(TargetArchitecture)</RuntimeIdentifier>
     <AppName>HelloiOS</AppName>
+    <GenerateXcodeProject>true</GenerateXcodeProject>
     <UseNativeAOTRuntime Condition="'$(UseNativeAOTRuntime)' == ''">true</UseNativeAOTRuntime>
+    <RunAOTCompilation>false</RunAOTCompilation>
     <NativeLib>static</NativeLib>
     <CustomNativeMain>true</CustomNativeMain>
     <!-- Escape NativeAOT bundling targets -->
     <NativeCompilationDuringPublish>false</NativeCompilationDuringPublish>
     <IlcCompileDependsOn>Compile;ComputeIlcCompileInputs;SetupOSSpecificProps;PrepareForILLink</IlcCompileDependsOn>
   </PropertyGroup>
 
+  <Import Project="$(MonoProjectRoot)\msbuild\apple\build\AppleBuild.props" />
+  <Import Project="$(MonoProjectRoot)\msbuild\apple\build\AppleBuild.InTree.targets" />
+
```

Comment:
Do these changes work when the sample is tested with `PublishAotUsingRuntimePack=true`?

Reply:
Let's check it out.

Reply:
There is an issues with the dependencies when using the runtime packs.
```
dotnet.sh \
	publish -c Release \
	-p:TargetOS=ios \
	-p:TargetArchitecture=arm64 \
	-p:DeployAndRun=false \
	-p:StripDebugSymbols=false \
	-p:PublishAotUsingRuntimePack=true \
	-bl
MSBuild version 17.7.0+5785ed5c2 for .NET
  Determining projects to restore...
  Restored /Users/miloskotlar/dotnet/runtime-ios/src/mono/sample/iOS-NativeAOT/Program.csproj (in 1,92 sec).
  Used runtime pack: /Users/miloskotlar/dotnet/runtime-ios/artifacts/bin/microsoft.netcore.app.runtime.ios-arm64/Release/
  Program -> /Users/miloskotlar/dotnet/runtime-ios/src/mono/sample/iOS-NativeAOT/bin/ios-arm64/Program.dll
  Generating native code
EXEC : error : Failed to load assembly 'System.Private.StackTraceMetadata' [/Users/miloskotlar/dotnet/runtime-ios/src/mono/sample/iOS-NativeAOT/Program.csproj]
  Internal.TypeSystem.TypeSystemException+FileNotFoundException: Failed to load assembly 'System.Private.StackTraceMetadata'
     at Internal.TypeSystem.ThrowHelper.ThrowFileNotFoundException(ExceptionStringID, String) + 0x3c
     at ILCompiler.Program.CreateInitializerList(CompilerTypeSystemContext) + 0x240
     at ILCompiler.Program.Run() + 0xe44
```

The aotsdk managed libraries are not included in the `.rsp` file. The changes in this PR shouldn't affect the AOT compilation but app bundling. I will compare results with the main branch.

Reply:
Thanks for checking out. I've updated the AppleBuild.InTree.targets to skip local runtime packs when the PublishAotUsingRuntimePack is used. The sample app is working with and without the PublishAotUsingRuntimePack property.

Reply:
Awesome, thank you!